### PR TITLE
 provider/openstack: allow providing user data to server group create/clone

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@derekolk @varikin @spinnaker/reviewers

**Comment by drmaas** [comment]
This is a prerequisite for external load balancers with consul.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy [hunk 1]
```diff
@@ -147,9 +148,22 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<DeploymentResult
       Subnet subnet = provider.getSubnet(description.region, subnetId)
       task.updateStatus BASE_PHASE, "Found network id $subnet.networkId from subnet $subnetId."
 
+      String userData
+      if (description.userData) {
+        if (description.userData.startsWith("http")) {
+          task.updateStatus BASE_PHASE, "Resolving user data from url $description.userData..."
+          userData = description.userData.toURL()?.text
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy]: What's the value in doing this?

**Comment by lwander** [comment]
Why is this needed for consul support?

**Comment by drmaas** [comment]
I was assuming the user needed to provide the script to boot consul, based on how consul was baked into the image. Are you saying that the startup script will be auto-generated based on clouddriver.yml config?

User data can also be used to execute other arbitrary startup functions, so I don' think the need goes away, even if the answer to the above is an auto-generated consul startup script.

**Comment by lwander** [comment]
Ideally the image should take care of running consul, but providing a user-script is definitely one way of doing this. It's probably better practice to make an upstart entry for the consul agent to handle crashes & ensure the agent is always running. But Spinnaker shouldn't take a hard stance on how to run Consul, so this is good to add.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy (hunk 1 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy]: Fetching scripts from an object store, e.g. http://docs.openstack.org/developer/swift

**Comment by lwander** [comment]
LGTM

**Pull request merged**